docs/internals/optimizer.rst

```diff
@@ -853,19 +876,29 @@ It tries to match patterns like ``X + 0`` on each subexpression.
 During the matching procedure, it resolves variables to their currently
 assigned expressions to be able to match more deeply nested patterns
 even when the code is in pseudo-SSA form.
+In fact, running CommonSubexpressionEliminator before helps this component track equivalent subexpressions.
 
 Some of the patterns like ``X - X -> 0`` can only be applied as long
 as the expression ``X`` is movable, because otherwise it would remove its potential side-effects.
 Since variable references are always movable, even if their current
 value might not be, the ExpressionSimplifier is again more powerful
-in split or pseudo-SSA form.
+in :ref:`expression-split <expression-splitter>` or :ref:`pseudo-SSA <ssa-transform>` form.
+
+Some of the simplifications destroy the :ref:`expression-split <expression-splitter>` form.
 
 .. _literal-rematerialiser:
 
 LiteralRematerialiser
 ^^^^^^^^^^^^^^^^^^^^^
 
-To be documented.
+If a variable referenced in an expression is known to have a literal value, replaces that use of
+the variable with the literal.
+
+This is mostly used so that other components do not have to rely on the DataflowAnalyzer.
+
+While the step destroys the :ref:`expression-split <expression-splitter>` form in the strict sense,
+it does not introduce any nested function calls, which may be good enough for many steps that require
+this property.
```

Comment:
Also added this for `LiteralRematerialiser`, based on our discussion on the chat.

Reply:
Does any step actually require the ast to be in expression split form as prerequisite? Isn't it rather that some steps run potentially more efficiently if it is fulfilled? In that case I would phrase it more like

```suggestion
While the step destroys the :ref:`expression-split <expression-splitter>` form in the strict sense,
it does not introduce any nested function calls, which may still be good enough for many steps that require
this property to function efficiently.
```

Reply:
Impacts at least `EqualStoreEliminator`